CRM/Core/Payment/GoCardless.php

```diff
@@ -98,29 +98,29 @@ public function buildForm(&$form) {
   /**
    * Attempt to cancel the subscription at GoCardless.
    *
-   * @see supportsCancelRecurring()
+   * @param \Civi\Payment\PropertyBag $propertyBag
    *
-   * @param string $message
-   * @param array $params
-   * which refers to 'subscriptionId'
-   *
-   * @return bool
+   * @return array|null[]
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
-  public function cancelSubscription(&$message = '', $params = []) {
-    if (empty($params['subscriptionId'])) {
-      throw new PaymentProcessorException("cancelSubscription requires a subscriptionId");
+  public function doCancelRecurring(\Civi\Payment\PropertyBag $propertyBag) {
```

Comment:
Ah, because the variable type `\Civi\Payment\PropertyBag` won't exist? By setting <ver>5.24</ver> in the info.xml you won't be given the option to upgrade if your CiviCRM is older than that). You can still do it manually (copy files etc.) but I guess you should know what you are doing then?

Reply:
Ah, no. \Civi\Payment\PropertyBag was available from 5.21. It was 5.25 when `doCancelRecurring()` was added to `CRM_Core_Payment`. So we should be fine?

Reply:
5.25 added the method `doCancelRecurring()` so it has only ever received a propertyBag. Before that `cancelSubscription()` was called.